### PR TITLE
bugfix: make sure of the folder structure to download the backup

### PR DESCRIPTION
**Pull request by rafaelcascalho**
This is a `bugfix` to issue #1 . Since it's internal, there's no need to alter the docs.

### CODE REVIEW

File: lib/morpheus-heroku/fetch.rb [hunk 1]
```diff
@@ -14,7 +14,28 @@ def generate_backup!
     end
 
     def download_backup!
+      prepare_location()
```
Comment (wenderjean) [inline lib/morpheus-heroku/fetch.rb]: Please, do not use parenthesis like that, it isn't common practice in Ruby.

File: lib/morpheus-heroku/fetch.rb [hunk 2]
```diff
@@ -14,7 +14,28 @@ def generate_backup!
...
       Helper.bash_run(command: "curl -o #{MorpheusHeroku.configuration.backup_location} `heroku pg:backups public-url`")
     end
+
+    def prepare_location
+      location = MorpheusHeroku.configuration.backup_location
+      unless File.writable?(location)
```
Comment (wenderjean) [inline lib/morpheus-heroku/fetch.rb]: What do you think to use a guard-clause instead?

```ruby
return if File.writable?(location)
```

File: lib/morpheus-heroku/fetch.rb [hunk 3]
```diff
@@ -14,7 +14,28 @@ def generate_backup!
...
+        prepare_dir(location[/^.+(?=\/)/])
+        prepare_file(location[/(?<=\/).+$/])
+      end
+    end
+
+    def prepare_dir(dir_name)
+      unless Dir.exist?(dir_name)
```
Comment (wenderjean) [inline lib/morpheus-heroku/fetch.rb]: You can use a guard-clause here also

```ruby
return if Dir.exist?(dir_name)
```

File: lib/morpheus-heroku/fetch.rb [hunk 4]
```diff
@@ -14,7 +14,28 @@ def generate_backup!
...
+        Helper.bash_run(command: "mkdir #{dir_name}")
+      end
+    end
+
+    def prepare_file(file_name)
+      unless File.exist?(file_name)
```
Comment (wenderjean) [inline lib/morpheus-heroku/fetch.rb]: The same here about guard-clauses

File: lib/morpheus-heroku/fetch.rb [hunk 5]
```diff
@@ -14,7 +14,28 @@ def generate_backup!
...
+
+    def prepare_location
+      location = MorpheusHeroku.configuration.backup_location
+      unless File.writable?(location)
+        prepare_dir(location[/^.+(?=\/)/])
+        prepare_file(location[/(?<=\/).+$/])
+      end
+    end
+
+    def prepare_dir(dir_name)
```
Comment (wenderjean) [inline lib/morpheus-heroku/fetch.rb]: Do you think `create_dir` would be a better name?

File: lib/morpheus-heroku/fetch.rb [hunk 6]
```diff
@@ -14,7 +14,28 @@ def generate_backup!
...
+      unless Dir.exist?(dir_name)
+        Helper.bash_run(command: "mkdir #{dir_name}")
+      end
+    end
+
+    def prepare_file(file_name)
```
Comment (wenderjean) [inline lib/morpheus-heroku/fetch.rb]: Do you think `create_file` would be a better name?

**Review by wenderjean — APPROVED** [review]
Looks good to me, good job.

**Comment by cionescu** [comment]
Great work @rcsm @wenderjean

**Pull request merged**